lib/phoenix/template.ex

```diff
@@ -132,6 +132,7 @@ defmodule Phoenix.Template do
       By default it raises but can be customized
       to render a particular template.
       """
+      @spec template_not_found(Template.name, map) :: no_return
```

Comment:
It would be best to use `Phoenix.Template.name`. This way definitely works but, since it depends on the alias defined at the top, it is a little brittle (and we won't fail to compile if a typespec is wrong). So using the full name is more guaranteed. Thank you :)